pkg/deviceplugin/server.go

```diff
@@ -160,13 +160,8 @@ func (srv *server) Allocate(ctx context.Context, rqt *pluginapi.AllocateRequest)
 				cresp.Mounts = append(cresp.Mounts, &dev.mounts[i])
 			}
 
-			for key, value := range dev.envs {
-				if cresp.Envs == nil {
-					cresp.Envs = make(map[string]string)
-				}
-
-				cresp.Envs[key] = value
-			}
+			cresp.Envs = dev.envs
+			cresp.Annotations = dev.annotations
```

Comment:
IIRC this loop was needed to copy values explicitly from one map to another. Otherwise `postAllocate()` could update `srv`'s state accidentally instead of the response.

Probably the slices above could be appended with mounts and nodes using the variadic syntax to save up a couple of branches, e.g.
```
m = append(m, n...)
```

Reply:
@rojkov 
> m = append(m, n...)

That would require changing types of srv.devices and srv.mounts I guess.

How about initializing Envs and Annotations unconditionally, e.g.
```
-                       cresp.Envs = dev.envs
-                       cresp.Annotations = dev.annotations
+                       cresp.Envs = map[string]string{}
+
+                       for key, value := range dev.envs {
+                               cresp.Envs[key] = value
+                       }
+
+                       cresp.Annotations = map[string]string{}
+
+                       for key, value := range dev.annotations {
+                               cresp.Annotations[key] = value
+                       }
```

Reply:
Yep, sounds good.

Reply:
done. Thank you for the review!